packages/notifications/push/amplify_push_notifications/lib/src/amplify_push_notifications_impl.dart

```diff
@@ -181,23 +172,47 @@ class AmplifyPushNotifications extends PushNotificationsPluginInterface {
     await _hostApi.setBadgeCount(badgeCount);
   }
 
+  Future<void> _registerDeviceWhenConfigure() async {
+    late String deviceToken;
+
+    try {
+      deviceToken = await onTokenReceived.first;
```

Comment:
`deviceToken` needs to declared earlier and initialized later because I moved the `_registerDevice` call after the catch block because it throws its own Exception if any.